commit 2e1eeb01cfcab78a2f79e7ec688e70db47f5c52a
Author: Stephan Mueller <smueller@chronox.de>
Date:   Fri May 26 12:11:31 2017 +0200

    crypto: drbg - Fixes panic in wait_for_completion call
    
    commit b61929c654f2e725644935737c4c1ea9c741e2f8 upstream.
    
    Initialise ctr_completion variable before use.
    
    Cc: <stable@vger.kernel.org>
    Signed-off-by: Harsh Jain <harshjain.prof@gmail.com>
    Signed-off-by: Stephan Mueller <smueller@chronox.de>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/crypto/drbg.c b/crypto/drbg.c
index 123d211..8cac3d3 100644
--- a/crypto/drbg.c
+++ b/crypto/drbg.c
@@ -1691,6 +1691,7 @@ static int drbg_init_sym_kernel(struct drbg_state *drbg)
 		return PTR_ERR(sk_tfm);
 	}
 	drbg->ctr_handle = sk_tfm;
+	init_completion(&drbg->ctr_completion);
 
 	req = skcipher_request_alloc(sk_tfm, GFP_KERNEL);
 	if (!req) {
